### PR TITLE
Rebrand my impact (only tree counter)

### PR DESCRIPTION
**Pull request by vicegax**
Gargantuan PR for tree counter on Your Impact

### CODE REVIEW

File: Client/Ecosia/UI/MyImpact/MyImpactCell.swift [hunk 1]
```diff
@@ -70,187 +136,104 @@ final class MyImpactCell: UICollectionViewCell, AutoSizingCell, Themeable {
...
-            bottomStack.isHidden = false
-            bottomStack.display(bottom)
+    
+    func update(personalCounter: Int) {
+        let progress = .init(personalCounter % 45) / 45.0
+        currentProgress.value = progress
+        indicator.value = progress
+        
+        if #available(iOS 15.0, *) {
+            treesCount.text = User.shared.impact.formatted()
```
Comment (ecotopian) [inline Client/Ecosia/UI/MyImpact/MyImpactCell.swift:202]: neat to leverage modern APIs where beneficial 🥇

File: Client/Ecosia/UI/SemanticColor.swift [hunk 2]
```diff
@@ -57,11 +61,15 @@ extension UIColor {
             static let secondary = UIColor(red: 0.871, green: 0.871, blue: 0.851, alpha: 1)
         }
 
-        public struct State {
```
Comment (ecotopian) [inline Client/Ecosia/UI/SemanticColor.swift:60]: 🧹

File: Client/Ecosia/UI/SemanticColor.swift [hunk 3]
```diff
@@ -34,6 +34,10 @@ extension UIColor {
             static let primary = UIColor(red: 0.059, green: 0.059, blue: 0.059, alpha: 1)
         }
         
+        struct Brand {
+            static let primary = UIColor(red: 0, green: 0.5, blue: 0.033, alpha: 1)
```
Comment (ecotopian) [inline Client/Ecosia/UI/SemanticColor.swift:38]: I also thought of adding this. Probably good to have it separately, though Button.primary is the same color 🎨

File: Client/Ecosia/UI/MyImpact/MyImpactCell.Progress.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        
+        required init?(coder: NSCoder) { nil }
+        
+        init() {
+            super.init(frame: .zero)
+            translatesAutoresizingMaskIntoConstraints = false
+            widthAnchor.constraint(equalToConstant: 240).isActive = true
+            heightAnchor.constraint(equalToConstant: 150).isActive = true
+            (layer as! CAShapeLayer).fillColor = UIColor.clear.cgColor
```
[the remark this answers is not in the document]
Comment (vauxhall) [reply Client/Ecosia/UI/MyImpact/MyImpactCell.Progress.swift:26]: I see what you mean, though I find having a property to cast another property but both with similar names a bit unpleasant. Might change in the future though, thanks for the suggestion.

File: Client/Ecosia/UI/MyImpact/MyImpactCell.swift [hunk 5]
```diff
@@ -3,63 +3,129 @@
...
+        
+        let treesCount = UILabel()
+        treesCount.translatesAutoresizingMaskIntoConstraints = false
+        treesCount.font = .systemFont(ofSize: UIFont.preferredFont(forTextStyle: .title1).pointSize, weight: .bold)
+        self.treesCount = treesCount
+        outline.addSubview(treesCount)
+        
+        let treesPlanted = UILabel()
+        treesPlanted.translatesAutoresizingMaskIntoConstraints = false
+        treesPlanted.font = .preferredFont(forTextStyle: .body)
```
[the remark this answers is not in the document]
Comment (vauxhall) [reply Client/Ecosia/UI/MyImpact/MyImpactCell.swift:60]: Good catch

**Pull request merged**